app/queries/hackney_api.rb

```diff
@@ -1,10 +1,16 @@
 class HackneyApi
+  class MissingResults < StandardError; end
+
   def initialize(json_api = JsonApi.new)
     @json_api = json_api
   end
 
   def list_properties(postcode:)
-    @json_api.get('v1/properties?postcode=' + postcode)
+    response = @json_api.get('v1/properties?postcode=' + postcode)
+
+    raise MissingResults unless response.is_a?(Hash) && response.key?('results')
+
+    response['results']
```

Comment:
Did you consider using `fetch`? That raises a `KeyError` if the key is not found.

Reply:
:+1: